File: app/src/main/java/com/kaigarrott/stopwatchkt/StopwatchService.java
package com.kaigarrott.stopwatchkt;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.os.IBinder;

public class StopwatchService extends Service {

    private static final String CHANNEL_ID = "stopwatch_channel";
    private Notification notification;

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
       // return super.onStartCommand(intent, flags, startId);
        if(notification == null) notification = setupNotification();
        startForeground(123, notification);
        return Service.START_STICKY;
    }

    private Notification setupNotification() {

        Notification.Builder builder;
        NotificationManager manager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);

        if(android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, "Stopwatch",
                    NotificationManager.IMPORTANCE_HIGH);
            channel.setDescription("Live stopwatch times");
            manager.createNotificationChannel(channel);
            builder = new Notification.Builder(getApplicationContext(), CHANNEL_ID);
        } else {
            builder = new Notification.Builder(getApplicationContext());
        }
        return builder.setContentTitle("Stopwatch")
                .setOngoing(true)
                .build();
    }
}

File: app/src/main/java/com/kaigarrott/stopwatchkt/StopwatchService.kt
package com.kaigarrott.stopwatchkt

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.Service
import android.content.Context
import android.content.Intent
import android.os.IBinder

class StopwatchService : Service() {
private var notification: Notification? = null

        override fun onBind(intent: Intent): IBinder? {
        return null
        }

        override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        if (notification == null) notification = setupNotification()
        startForeground(123, notification)
        return START_STICKY
        }

private fun setupNotification(): Notification {

        val builder: Notification.Builder
        val manager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager

        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
        val channel = NotificationChannel(CHANNEL_ID, "Stopwatch",
        NotificationManager.IMPORTANCE_HIGH)
        channel.description = "Live stopwatch times"
        manager.createNotificationChannel(channel)
        builder = Notification.Builder(applicationContext, CHANNEL_ID)
        } else {
        builder = Notification.Builder(applicationContext)
        }
        return builder.setContentTitle("Stopwatch")
        .setOngoing(true)
        .build()
        }

        companion object {

private val CHANNEL_ID = "stopwatch_channel"
        }
        }
